src/Framework/AspNetCoreAnalyzers/src/CodeFixes/WebApplicationBuilderFixer.cs

```diff
@@ -0,0 +1,165 @@
+     });
+
+    public sealed override FixAllProvider GetFixAllProvider() => WellKnownFixAllProviders.BatchFixer;
+
+    public sealed override Task RegisterCodeFixesAsync(CodeFixContext context)
+    {
+        foreach (var diagnostic in context.Diagnostics)
+        {
+            context.RegisterCodeFix(
+                CodeAction.Create("Fix Configure methods",
+                    cancellationToken => FixConfigureMethods(diagnostic, context.Document, cancellationToken),
+                    equivalenceKey: DiagnosticDescriptors.DisallowConfigureAppConfigureHostBuilder.Id),
+                diagnostic);
+        }
+
+        return Task.CompletedTask;
+    }
+
+    private static async Task<Document> FixConfigureMethods(Diagnostic diagnostic, Document document, CancellationToken cancellationToken)
+    {
+        // hey Safia some of the below are just draft comments to help you understand the code easier.
+        // We're gonna delete it later before the final commit
+        
+        DocumentEditor editor = await DocumentEditor.CreateAsync(document, cancellationToken);
+        var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+
+        if (root == null)
+        {
+            return document;
+        }
+
+        var invocation = root.FindNode(diagnostic.Location.SourceSpan); //whole invocation starting from "builder"
+      
+        if (invocation is InvocationExpressionSyntax { } newInvocationExpr)
+        {
+            if (newInvocationExpr.Expression is not MemberAccessExpressionSyntax hostBasedInvocationMethodExpr)
+            {
+                return editor.OriginalDocument;
+            }
+
+            var configuration = SyntaxFactory.IdentifierName("Configuration");
+
+            if (hostBasedInvocationMethodExpr.Expression is not MemberAccessExpressionSyntax subExpression)
+            {
+                return editor.OriginalDocument;
+            }
+            if (subExpression.Name != null) //subExpression.Name is Host/Webhost
+            {
+                subExpression = subExpression.WithName(configuration);
+                hostBasedInvocationMethodExpr = hostBasedInvocationMethodExpr.WithExpression(subExpression); //becomes builder.Configuration.ConfigureAppConfiguration
+            }
+
+            if (newInvocationExpr.ArgumentList.Arguments.Count == 0)
+            {
+                return editor.OriginalDocument;
+            }
+
+            var initArgumentExpr = newInvocationExpr.ArgumentList.Arguments[0].Expression; // builder => { }
+
+            if (initArgumentExpr is ParenthesizedLambdaExpressionSyntax { } parenLambdaExpr)
+            {
+                var bodyEnum = ExtractParenLambdaBody(parenLambdaExpr).GetEnumerator();
+                while (bodyEnum.MoveNext())
+                {
+                    if (bodyEnum.Current is not ExpressionStatementSyntax currentStatement) //builder.AddJsonFile() or builder.AddEnvironmentVariables()
```

Comment:
We can combine this if-statement and the one below into one statement with an `&&`.